plugins/UsersManager/API.php

```diff
@@ -769,9 +771,10 @@ public function inviteUser($userLogin, $email, $initialIdSite = null, $expiryInD
         UsersManager::dieIfUsersAdminIsDisabled();
 
         // check password confirmation only when using session auth
-        if (Common::getRequestVar('force_api_session', 0)) {
-            $this->confirmCurrentUserPassword($passwordConfirmation);
-        }
+        //notes as design request this is removed.
+//        if (Common::getRequestVar('force_api_session', 0)) {
+//            $this->confirmCurrentUserPassword($passwordConfirmation);
+//        }
```

Comment:
I know right, but the designer asks me to remove this.

Reply:
Sacrificing security for usability reasons isn't anything we should consider. Did you let @Javi-Ormaechea know, that this will make Matomo bit less secure? If that's a mandatory requirement you should at least let @tsteur confirm that.

Reply:
From a security perspective, to invite a user via the UI we need to ask for a password confirmation for security purposes. 

I don't know where exactly we're asking for a password currently in the process. If it helps it could also look different. For example, that we only redirect to the "Invite user" screen after a password confirmation on a dedicated page. But then a user has only like 15 minutes to complete the action.

Reply:
@tsteur I was not aware that this was a security issue but I'm curious to know more about why a password needs to be added here when the user has technically logged in already. As an experience it makes it less seamless, and probably unexpected too (_last bit is an assumption_).

Reply:
@Javi-Ormaechea for sensitive actions we require a password confirmation as otherwise someone else could go onto your computer/laptop/tablet/... where you are logged in already and add a user on your behalf without your knowing. I don't know what the current screen looks like and if we ask for the password in the form. 

In some places we redirect the user for example first to a dedicated page for confirming the password and then redirect to the actual page. This password confirmation is then valid for I believe 15 minutes or so. Meaning there are options to change this.

Reply:
@tsteur I can't see the password confirmation remembered for 15 minutes.